WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/CouponExpiryDateView.swift

```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+import WordPressAuthenticator
+
+/// View for selecting a date in SwiftUI.
+///
+struct CouponExpiryDateView: View {
+
+    @State var date: Date = Date()
+    let completion: ((Date) -> Void)
+
+    var body: some View {
+        GeometryReader { geometry in
+            ScrollView {
+                VStack {
+                    DatePicker("Date picker", selection: $date, displayedComponents: .date)
+                        .datePickerStyle(GraphicalDatePickerStyle())
+                        .onChange(of: date) { newDate in
+                            completion(newDate)
+                        }
+                    Spacer()
+                }
+            }
+        }
+        .ignoresSafeArea(.container, edges: [.horizontal, .bottom])
+        .navigationTitle(Localization.title)
+    }
+}
+
+// MARK: - Constants
+//
+private extension CouponExpiryDateView {
+
+    enum Localization {
+        static let title = NSLocalizedString("Select an expiry date",
```

Comment:
Suggestion: should the title be more concise?
```suggestion
        static let title = NSLocalizedString("Coupon expiry date",
```